mathbot/modules/dice.py

```diff
@@ -26,7 +36,81 @@ class DiceModule(core.module.Module):
 		faces = int(faces or 6)
 		if faces <= 0:
 			return '🎲 Dice must have a positive number of faces.'
-		if faces > 100000 or dice > 100000:
-			return '🎲 Values are too large. Cannot be greater than 100000.'
-		total = sum(random.randint(1, faces) for i in range(dice))
-		return f'🎲 {total}'
+
+		limit = await self.get_limit(message)
+
+		# this is the minimal length of this query, it is used to determine
+		# whether it's possible for the result to be short enough to fit
+		# within the limit.
+		#
+		# I got this by assuming each die rolled 1, plus 1 space per die
+		# giving me 2 * dice. Then i add the length of the total, and the
+		# length of the extra stuff that's always in the result.
+		min_len = 2 * dice + 9 + math.log10(dice)
+
+		# gaussian roll is faster so try that if we can't show all the rolls
+		if min_len >= limit:
+			total = 0
+			try:
+				total = self.gaussian_roll(dice, faces)
+			except ValuesTooBigException:
+				return '🎲 Values are too large.'
+
+			return f'🎲 total: {total}'
+		else:
+			rolls, total = self.formatted_roll(dice, faces)
+			final_message = f'🎲 {rolls}'
+			return final_message if len(final_message) <= limit else f'🎲 total: {total}'
+
+	async def get_limit(self, message):
+		'''This method gets the character limit for messages.'''
+		unlimited = await core.settings.resolve_message('f-roll-unlimited', message)
+		return 200 if not unlimited else 2000
+
+	def formatted_roll(self, dice, faces):
+		'''
+		This will roll dice and return a string of the results as well as the total.
+		'''
+		rolls = [random.randint(1, faces) for _ in range(dice)]
+		total = sum(rolls)
+		s = f'{str.join(" ", map(str, rolls))} (total: {total})'
+		return s, total
+
+	def gaussian_roll(self, dice, faces, limit=100000):
+		'''[random.randint(1, faces) for _ in range(dice)]
+		This method simulates a roll using normal distributions. It'll do it as
```

Comment:
`Simulate a roll using normal distributions. Do it as many times as necessary to avoid float inaccuracy, unless that means rolling more times than limit.`